### PR TITLE
[prometheus-druid-exporter] Add securityContext and containerSecurity…

### PR DESCRIPTION
**Pull request by JulioZanette**
Add securityContext and containerSecurityContext

Signed-off-by: Zanette, Julio <juliozanette@outlook.com>

#### What this PR does / why we need it:
This PR adds securityContext and containerSecurityContext to Prometheus Druid Exporter
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by gkarthiks — COMMENTED** [review]
@JulioZanette can you please sign your commits?

**Comment by JulioZanette** [comment]
> @JulioZanette can you please sign your commits?

Signed!

**Comment by JulioZanette** [comment]
> @JulioZanette can you please sign your commits?

Signed

**Review by monotek — CHANGES_REQUESTED** [review]
please fix ci

**Comment by gkarthiks** [comment]
not sure why this is failing may be @iamabhishek-dubey could help this PR. 


I am not an expert with this chart, but the logs show that the `wget` fails [here](https://github.com/prometheus-community/helm-charts/pull/1018/checks?check_run_id=2696220028#step:9:201) may be that's the reason?

**Comment by JulioZanette** [comment]
> not sure why this is failing may be @iamabhishek-dubey could help this PR.
> 
> I am not an expert with this chart, but the logs show that the `wget` fails [here](https://github.com/prometheus-community/helm-charts/pull/1018/checks?check_run_id=2696220028#step:9:201) may be that's the reason?

Same on my side, I cannot see what is wrong with the `wget` test. It all works as expected on my side.

**Comment by iamabhishek-dubey** [comment]
I will look into this

**Comment by iamabhishek-dubey** [comment]
Can you please try to change this line to

https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-druid-exporter/templates/tests/connection-test.yaml#L18

```yaml
args:  ['-qO-', '{{ include "prometheus-druid-exporter.fullname" . }}.svc.cluster.local:{{ .Values.exporterPort }}/metrics']
```

**Review by monotek — APPROVED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
When trying out the proposed change of @iamabhishek-dubey please also merge main into your branch to get the "node not ready" problem in ci fixed.

**Comment by gkarthiks** [comment]
hi @JulioZanette can you give it a shot?

**Comment by JulioZanette** [comment]
Hi @gkarthiks apologize for the late response, been busy these days.

I have applied the patch suggested by @iamabhishek-dubey to connection-test.yaml and merged main into my branch as suggested by @monotek and committed changes.

**Comment by gkarthiks** [comment]
hi @JulioZanette thanks, it looks good to me. I cannot merge it, but @iamabhishek-dubey can you please help him here?

**Comment by iamabhishek-dubey** [comment]
Looks like the internal DNS name is not resolving .cluster.local, can you give a try with

```yaml
args:  ['-qO-', '{{ include "prometheus-druid-exporter.fullname" . }}.svc:{{ .Values.exporterPort }}/metrics']
```

I am pretty much sure it should work because .svc is part of the FQDN in coredns/kube-dns

**Comment by JulioZanette** [comment]
Applied  @iamabhishek-dubey suggestion.

File: charts/prometheus-druid-exporter/templates/tests/connection-test.yaml [hunk 1]
```diff
@@ -15,5 +15,6 @@ spec:
   - name: wget
     image: busybox
     command: ['wget']
-    args:  ['-qO-', '{{ include "prometheus-druid-exporter.fullname" . }}:{{ .Values.exporterPort }}/metrics']
+    args:  ['-qO-', '{{ include "prometheus-druid-exporter.fullname" . }}.svc:{{ .Values.exporterPort }}/metrics']
```
Comment (monotek) [inline charts/prometheus-druid-exporter/templates/tests/connection-test.yaml:18]: When dns is used the namesapce needs to be given too.
See: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services

So i guess it should look like:

```suggestion
    args:  ['-qO-', '{{ include "prometheus-druid-exporter.fullname" . }}.{{ .Release.Namespace }}.svc:{{ .Values.exporterPort }}/metrics']
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-druid-exporter/templates/tests/connection-test.yaml (hunk 1 above)
Comment (JulioZanette) [reply charts/prometheus-druid-exporter/templates/tests/connection-test.yaml]: Updated and pushed. Waiting for someone for pipeline to run.

Comment (gkarthiks) [reply charts/prometheus-druid-exporter/templates/tests/connection-test.yaml]: hi @JulioZanette, I re-ran the CI and all your changes are working fine. I cannot merge the PR. 

You gotta check with @iamabhishek-dubey or @sandy724 to get this merged.

**Review by monotek — APPROVED** [review]

**Pull request merged**